src/Nix/Value.hs

```diff
@@ -142,6 +141,14 @@ data NValueF p m r
 
 -- ** Eq1
 
+instance Eq r => Eq (NValueF p m r) where
+  (==) (NVConstantF x) (NVConstantF y) = x == y
+  (==) (NVStrF      x) (NVStrF      y) = x == y
+  (==) (NVPathF     x) (NVPathF     y) = x == y
+  (==) (NVListF     x) (NVListF     y) = x == y
+  (==) (NVSetF  _   x) (NVSetF _    y) = x == y
+  (==) _               _               = False
+
 instance Eq1 (NValueF p m) where
   liftEq _  (NVConstantF x) (NVConstantF y) = x == y
   liftEq _  (NVStrF      x) (NVStrF      y) = x == y
```

Comment:
Great, fully lawful `Eq`.

Notice here, that there was a `-- ** Eq1`, which is a sectioning of the code, that helps to partition, organize, have the high view, and navigate through the code as a tree structure:

This module code structure is:
![image](https://github.com/haskell-nix/hnix/assets/20933385/896dc979-0fe4-474b-bf8a-ccdd271c6e28)

I believe this is supported in several editors nowadays. I used Emacs `outshine` at the time.
 
This is not that important to fix it to merge the PR, but to have a note that those are the code section denotations.

Reply:
Interesting. I personally prefer to do this sort of thing with code outline plugins based on syntax rather than comments. I prefer to have a single source of truth.

I've added the relevant section names.